drive/amazon/amazon.go

```diff
@@ -46,29 +47,30 @@ func NewClient(c drive.Config) (drive.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &AmazonCloudDrive{
+	return &Drive{
 		client: client,
 		ep:     ep,
 		config: c,
 		files:  make(map[string]string),
 	}, nil
 }
 
-type AmazonCloudDrive struct {
-	client *http.Client
-	ep     *Endpoint
-	config drive.Config
-	// files maps from the string([]byte) representation of the file's SHA2 to
-	// the corresponding fileID in CloudDrive
-	files        map[string]string
+// Drive is a representation of the Amazon Cloud storage system.
+type Drive struct {
 	sync.RWMutex // protects files
```

Comment:
I think it's more important to keep it next to the thing it protects.  If you feel strongly about not having anonymous structs up top, a good compromise might be to give the field a name, eg. "fl", "fm" or "flock" and add the few characters to the places it's used.